tests/test_iot/test_iot.py

```diff
@@ -1991,3 +1991,133 @@ def test_list_job_executions_for_thing():
+            {"dynamoDBv2": {"putItem": {"tableName": "my-table"}, "roleArn": "my-role"}}
+        ],
+        "errorAction": {
+            "republish": {"qos": 0, "roleArn": "my-role", "topic": "other-topic"}
+        },
+        "description": "my-description",
+        "ruleDisabled": False,
+        "awsIotSqlVersion": "2016-03-23",
+    }
+
+    # create
+    client.create_topic_rule(ruleName=name, topicRulePayload=payload)
+
+    # list
+    res = client.list_topic_rules()
+    res.should.have.key("rules").which.should.have.length_of(1)
+    for rule in res["rules"]:
+        rule.should.have.key("ruleName").which.should.equal(name)
+        rule.should.have.key("createdAt").which.should_not.be.none
+        rule.should.have.key("ruleArn").which.should_not.be.none
+        rule.should.have.key("ruleDisabled").which.should.equal(payload["ruleDisabled"])
+        rule.should.have.key("topicPattern").which.should.equal("topic/*")
+
+    # get
+    rule = client.get_topic_rule(ruleName=name)
+    rule.should.have.key("ruleArn").which.should_not.be.none
+    rule.should.have.key("rule")
+    rrule = rule["rule"]
+    rrule.should.have.key("actions").which.should.equal(payload["actions"])
+    rrule.should.have.key("awsIotSqlVersion").which.should.equal(
+        payload["awsIotSqlVersion"]
+    )
+    rrule.should.have.key("createdAt").which.should_not.be.none
+    rrule.should.have.key("description").which.should.equal(payload["description"])
+    rrule.should.have.key("errorAction").which.should.equal(payload["errorAction"])
+    rrule.should.have.key("ruleDisabled").which.should.equal(payload["ruleDisabled"])
+    rrule.should.have.key("ruleName").which.should.equal(name)
+    rrule.should.have.key("sql").which.should.equal(payload["sql"])
+
+    # replace
+    payload["description"] = "new-description"
+    client.replace_topic_rule(
+        ruleName=name, topicRulePayload=payload,
+    )
+    rule = client.get_topic_rule(ruleName=name)
+    rule["rule"]["ruleName"].should.equal(name)
+    rule["rule"]["description"].should.equal(payload["description"])
+
+    # disable
+    client.disable_topic_rule(ruleName=name)
+    rule = client.get_topic_rule(ruleName=name)
+    rule["rule"]["ruleName"].should.equal(name)
+    rule["rule"]["ruleDisabled"].should.equal(True)
+
+    # enable
+    client.enable_topic_rule(ruleName=name)
+    rule = client.get_topic_rule(ruleName=name)
+    rule["rule"]["ruleName"].should.equal(name)
+    rule["rule"]["ruleDisabled"].should.equal(False)
+
+    # delete
+    client.delete_topic_rule(ruleName=name)
+    res = client.list_topic_rules()
+    res.should.have.key("rules").which.should.have.length_of(0)
+
+    # errors
+
+    # get not existing rule
+    try:
+        client.get_topic_rule(ruleName=name)
+    except ClientError as exc:
+        error_code = exc.response["Error"]["Code"]
+        error_code.should.equal("ResourceNotFoundException")
+    else:
+        raise Exception("Should have raised error")
```

Comment:
Error handling should follow this format:
```
with pytest.raises(..):
    ....
````
Like it's done in an earlier test case: https://github.com/spulec/moto/blob/master/tests/test_iot/test_iot.py#L607

Reply:
@bblommers I've updated the tests implementation. Please check it out.